### PR TITLE
feat: pass parent beacon block root to payload builder

### LINKED ISSUE #4401: Pass parent beacon block root to payload builder
### Describe the feature

Currently we use `None` for the `parent_beacon_block_root` argument in the payload builder:
https://github.com/paradigmxyz/reth/blob/0c7a93717aa516f34e79ac6bafd8be088be8b7cc/crates/payload/basic/src/lib.rs#L788

The `PayloadAttributes` from `engine_forkchoiceUpdatedV3` contain a `parent_beacon_block_root` field:
https://github.com/paradigmxyz/reth/blob/0c7a93717aa516f34e79ac6bafd8be088be8b7cc/crates/rpc/rpc-types/src/eth/engine/payload.rs#L296-L300

This argument should be passed to the payload builder to populate this field.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alessandromazza98**
Closes #4401 

This PR should close that issue. I'm actually not 100% sure that it is complete. Please let me know.

### CODE REVIEW

File: crates/payload/builder/src/payload.rs [hunk 1]
```diff
@@ -129,6 +129,8 @@ pub struct PayloadBuilderAttributes {
     pub prev_randao: H256,
     /// Withdrawals for the generated payload
     pub withdrawals: Vec<Withdrawal>,
+    /// Root of the parent beacon block
+    pub parent_beacon_block_root: H256,
```
Comment (Rjected) [inline crates/payload/builder/src/payload.rs:133]: This needs to be an `Option<H256>` so header conversions can preserve `None` pre-cancun

**Review by Rjected — CHANGES_REQUESTED** [review]
Important note on making sure we keep the root as `None` pre-cancun, otherwise this looks good

**Review by mattsse — APPROVED** [review]
lgtm, pending @Rjected

**Review by Rjected — APPROVED** [review]
LGTM, thanks!

**Pull request merged**